### ISSUE TITLE
[Question] Git clone acceleration

### ISSUE THREAD

**Issue by joeky888** [facebook/sapling#572]
Is it possible to accelerate a git-cloning process? Or can I use sapling as an alternative to acclerate programing language package managers like `composer install`, `go get` etc.? Use cases are CI/CD and docker build. Thanks.

**Comment by zzl0** [comment]
Do you mean only use Sapling for cloning the git repo? do you need to integrate the cloned repo with other tools?

**Comment by joeky888** [comment]
That's right. I would like to use it for cloning only. If I can just do something like `alias git=sl` in CI to accelerate `go get` and `composer install`. When [bun](https://github.com/oven-sh/bun) comes out, I immediately switched from npm to bun in CI and the time spending on `bun install` is about only one-thrid of `npm install`. ON THE SAME HARDWARE! And this got me wondering why can't pacakge managers push their limits instead of wasting time on the sufficient bandwidth that doesn't make any sense to me.

You may also heard of [aria2](https://github.com/aria2/aria2) another tool to accelerate downloading speed. Anyways, nowadays a lot of package managers depends on git, it would be a killer feature if sapling can boost the cloning process. That said, if you snooze, you lose.
